mmdeploy/codebase/mmdet/models/dense_heads/condinst_head.py

```diff
@@ -0,0 +1,359 @@
+    device = bbox_preds[0].device
+    cfg = self.test_cfg if cfg is None else cfg
+    batch_size = bbox_preds[0].shape[0]
+    featmap_sizes = [cls_score.shape[-2:] for cls_score in cls_scores]
+    pre_topk = cfg.get("nms_pre", -1)
+
+    all_level_points_strides = self.prior_generator.grid_priors(
+        featmap_sizes, device=device, with_stride=True
+    )
+    all_level_points = [i[:, :2] for i in all_level_points_strides]
+    all_level_strides = [i[:, 2] for i in all_level_points_strides]
+
+    flatten_cls_scores = [
+        cls_score.permute(0, 2, 3, 1).reshape(batch_size, -1, self.cls_out_channels)
+        for cls_score in cls_scores
+    ]
+    flatten_bbox_preds = [
+        bbox_pred.permute(0, 2, 3, 1).reshape(batch_size, -1, 4)
+        for bbox_pred in bbox_preds
+    ]
+    flatten_score_factors = [
+        score_factor.permute(0, 2, 3, 1).reshape(batch_size, -1, 1)
+        for score_factor in score_factors
+    ]
+    flatten_param_preds = [
+        param_pred.permute(0, 2, 3, 1).reshape(batch_size, -1, self.num_params)
+        for param_pred in param_preds
+    ]
+    flatten_cls_scores = torch.cat(flatten_cls_scores, dim=1).sigmoid()
+    flatten_bbox_preds = torch.cat(flatten_bbox_preds, dim=1)
+    flatten_score_factors = torch.cat(flatten_score_factors, dim=1).sigmoid()
+    flatten_param_preds = torch.cat(flatten_param_preds, dim=1)
+
+    points = torch.cat(all_level_points)
+    strides = torch.cat(all_level_strides)
+    tl_x = points[..., 0] - flatten_bbox_preds[..., 0]
+    tl_y = points[..., 1] - flatten_bbox_preds[..., 1]
+    br_x = points[..., 0] + flatten_bbox_preds[..., 2]
+    br_y = points[..., 1] + flatten_bbox_preds[..., 3]
+    #
+    bboxes = torch.stack([tl_x, tl_y, br_x, br_y], -1)  # decode
+    scores = flatten_cls_scores
+    score_factors = flatten_score_factors
+    param_preds = flatten_param_preds
+    scores = scores * score_factors
+
+    # get post processing config
+    post_params = get_post_processing_params(deploy_cfg)
+    max_output_boxes_per_class = post_params.max_output_boxes_per_class
+    iou_threshold = cfg.nms.get("iou_threshold", post_params.iou_threshold)
+    score_threshold = cfg.get("score_thr", post_params.score_threshold)
+    pre_top_k = post_params.pre_top_k
+    keep_top_k = cfg.get("max_per_img", post_params.keep_top_k)
+    nms_type = cfg.nms.get("type")
+
+    # Rewrite nms for onnx convertion
+    if with_nms:
+        (
+            batched_dets,
+            batched_labels,
+            batched_param_preds,
+            batched_points,
+            batched_strides,
+        ) = _nms_condinst(
+            self,
+            bboxes,
+            scores,
+            score_factors,
+            param_preds,
+            points,
+            strides,
+            max_output_boxes_per_class,
+            iou_threshold,
+            score_threshold,
+            pre_top_k,
+            keep_top_k,
+            nms_type,
+        )
+
```

Comment:
we need to return output in `torch.Tensor` type, not InstanceData.

Reply:
I will modify it soon.